src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2161 @@
+
+    def get_patterning_mode(self) -> str:
+        """Returns: (str) the mode of the patterning."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.get_patterning_mode()
+
+    def set_patterning_mode(self, mode: str) -> None:
+        """Set the mode of the patterning. (Serial, Parallel)"""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            self.server.set_patterning_mode(mode)
+
+    def clear_patterns(self) -> None:
+        """Clear all patterns."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            self.server.clear_patterns()
+
+    def set_default_application_file(self, application_file: str = "Si") -> None:
+        """
+        Set the default application file.
+        :param application_file: (str) Name of the default application file.
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            self.server.set_default_application_file(application_file)
+
+    def set_default_patterning_beam_type(self, channel: str) -> None:
+        """
+        Set the default patterning beam type.
+        :param channel: (str) Name of one of the channels.
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            self.server.set_default_patterning_beam_type(channel)
+
+    def get_available_application_files(self) -> list:
+        """Returns: (list) the available application files."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.get_available_application_files()
+
+    def estimate_milling_time(self) -> float:
+        """Returns: (float) the estimated milling time for drawn patterns."""
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.estimate_milling_time()
+
+
+class Scanner(model.Emitter):
+    """
+    This is an extension of the model.Emitter class. It contains Vigilant
+    Attributes for magnification, accel voltage, blanking, spotsize, beam shift,
+    rotation and dwell time. Whenever one of these attributes is changed, its
+    setter also updates another value if needed.
+    """
+
+    def __init__(self, name: str, role: str, parent: SEM, hfw_nomag: float, channel: str, has_detector: bool=False, **kwargs):
+        """
+        :param name (str): name of the Scanner
+        :param role (str): role of the Scanner
+        :param parent (SEM): parent of the Scanner
+        :param hfw_nomag (float): horizontal field width at nominal magnification
+        :param channel (str): name of the electron channel
+        :param has_detector (bool): True if a Detector is also controlled. In this case,
+          the .resolution, .scale and associated VAs will be provided too.
+        """
+        self.parent: SEM  # for type hinting
```

Comment:
```suggestion
        self.parent: SEM = parent  # for type hinting
```

Reply:
Not sure of this is needed because in the `Component` class there is a setter and getter

```
    @property
    def parent(self):
        if self._parent:
            return self._parent()
        else:
            return None

    @parent.setter
    def parent(self, p):
        if p:
            assert isinstance(p, Component)
            self._parent = weakref.ref(p)
        else:
            self._parent = None

```